### PR TITLE
Add flag to switch off MC identifiers in MapDatasetEventSampler

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR introduces a flag to switch-off the tag of the sampled events of a given model with a Montecarlo identifier. A test has been added as well.

### CODE REVIEW

**Review by registerrier — APPROVED** [review]
Thanks @fabiopintore . This looks good. I have left 2 small inline comments.
My main suggestion would be to rename the argument `mc_id` with something more explicict like  `keep_mc_id` or `store_mic_id`

File: gammapy/datasets/simulate.py [hunk 1]
```diff
@@ -31,14 +31,22 @@ class MapDatasetEventSampler:
         an energy-dependent time-varying source
     t_delta : `~astropy.units.Quantity`
         Time interval used to sample the time-dependent source
+    mc_id : `~bool`
+        Flag to tag sampled events from a given model with a Montecarlo identifier. If set to "False", no
+        identifier will be assigned
```
Comment (bkhelifi) [inline gammapy/datasets/simulate.py:36]: Can you add the default value in the docstring?

File: gammapy/datasets/simulate.py [hunk 2]
```diff
@@ -387,7 +397,7 @@ def event_det_coords(observation, events):
         return events
 
     @staticmethod
-    def event_list_meta(dataset, observation):
+    def event_list_meta(dataset, observation, mc_id=True):
```
Comment (bkhelifi) [inline gammapy/datasets/simulate.py:400]: This new parameter should be described in the doctring

File: gammapy/datasets/simulate.py [hunk 3]
```diff
@@ -264,13 +272,14 @@ def sample_sources(self, dataset):
...
+            if self.mc_id:
+                if len(table) == 0:
+                    mcid = table.Column(name="MC_ID", length=0, dtype=int)
+                    table.add_column(mcid)
 
-            table["MC_ID"] = idx + 1
-            table.meta["MID{:05d}".format(idx + 1)] = idx + 1
-            table.meta["MMN{:05d}".format(idx + 1)] = evaluator.model.name
+                table["MC_ID"] = idx + 1
+                table.meta["MID{:05d}".format(idx + 1)] = idx + 1
```
[the remark this answers is not in the document]
Comment (bkhelifi) [reply gammapy/datasets/simulate.py:281]: I misunderstand what it means, sorry. all good then

**Review by bkhelifi — APPROVED** [review]
Thank you a lot @fabiopintore

**Pull request merged**